### PR TITLE
Split Project2015To2017 into Core and Migrate2017

### PR DESCRIPTION
**Pull request by andrew-boyarshin**
* Move almost everything to a new _Core_ project
* Move VS15-related transformations to _Migrate2017.Library_
* Make `ProjectConverter.Convert` an extension method (95% compatible, new optional/recommended `ILogger` parameter added)
* Retain umbrella-project _Project2015To2017_ to preserve NuGet package
* Move all common properties to `Directory.Build.props`
* Give Project2015To2017Tests a proper name with a dot
* Update external NuGet dependencies

P.S. don't worry about 191 changed files - that's mainly moved files.

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<Project>
```
Comment (hvanbakel) [inline Directory.Build.props]: Sweet! This was on my list for quite some time.

File: Project2015To2017.Core/Project2015To2017.Core.csproj [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFrameworks>netstandard1.3;netstandard2.0</TargetFrameworks>
+    <RootNamespace>Project2015To2017</RootNamespace>
```
Comment (hvanbakel) [inline Project2015To2017.Core/Project2015To2017.Core.csproj]: We don't need to do this now but I like having the namespaces line up with the project names.

Comment (andrew-boyarshin) [reply Project2015To2017.Core/Project2015To2017.Core.csproj]: Me too, but that's for maintaining (mostly) stable public API surface.

**Comment by andrew-boyarshin** [comment]
Since no issues are raised, I'll rebase this against latest master and merge this later today.

**Pull request merged**